src/modules/powerrename/lib/PowerRenameManager.cpp

```diff
@@ -903,22 +903,34 @@ DWORD WINAPI CPowerRenameManager::s_regexWorkerThread(_In_ void* pv)
                                     StringCchCopy(sourceName, ARRAYSIZE(sourceName), originalName);
                                 }
 
-                                PWSTR replaceTerm = nullptr;
                                 SYSTEMTIME fileTime = {0};
 
-                                if (!useFileTime || ( SUCCEEDED(spItem->GetTime(&fileTime)) && SUCCEEDED(spRenameRegEx->PutFileTime(fileTime))))
+                                HRESULT hr = S_OK;
+                                if (useFileTime)
+                                {
+                                    hr = spItem->GetTime(&fileTime);
+                                    if (SUCCEEDED(hr))
+                                    {
+                                        hr = spRenameRegEx->PutFileTime(fileTime);
+                                    }
+                                }
+
+                                PWSTR newName = nullptr;
+
+                                if (SUCCEEDED(hr))
```

Comment:
Checking here an HRESULT that was produced several lines of code before it's not a good idea, what if in the future we need to add another call in between that also returns an HRESULT?
You should handled the HRESULT error as soon as it happens and in case set another variable to provide that information to the code ahead.